packages/api/src/utils/billing-types.ts

```diff
@@ -58,6 +59,8 @@ export interface CustomersService {
 
 export type StoragePriceName = 'free' | 'lite' | 'pro'
 
+export type StoragePrice = StoragePriceName | Pick<Stripe.Response<Stripe.Price>, 'id'|'metadata'|'tiers'>
```

Comment:
Can you do this without depending on `stripe` package for types, that way this API can be implemented against non-stripe billing providers?

That way we can also hide stripe-specific implementation details from consumers of this HTTP API, e.g. from the frontend. i.e. in this PR the frontend code will throw runtime errors if the custom storage price object `metadata` property doesn't have keys for 'UI Label', 'Description', and 'Bandwidth'. We want  stripe-specific parsing happens in order to produce something in `billing-types` so that anything consuming billing-types (e.g. our frontend) is abstracted from stripe. It also allows us to keep the api json objects using consistent naming e.g. thisCase vs snake_case.

One way of doing that is to define a `CustomStoragePrice` type with eactly what you need:

```
type CustomStoragePrice = { 
  // in StripeSubscriptionsService, parse this from metadata['UI Label']
  name: string; 
  // in StripeSubscriptionsService, parse this from metadata['Description']
  description: string;
  // parse from stripe 'tiers' info
  tiers: Array<{ unit: 'byte', max: number, price: ['usd', number] }>
}

and in the response
```
price: StoragePriceName | CustomStoragePrice
```

and then no need to have this file import from npm package `stripe`.

Reply:
updated (see [here](https://github.com/web3-storage/web3.storage/pull/2152#discussion_r1065217803))